### PR TITLE
feat(byods): Spark 549034 jest setup

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-549034 

## This pull request addresses
Jest Setup for BYoDS SDK

## by making the following changes
Added basic jest config for byods package newly added and tested it with dummy test files.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/byods/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+export default {
+  preset: 'ts-jest',
+  rootDir: './',
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+  testEnvironment: 'jsdom',
```
Comment (bhabalan) [inline packages/byods/jest.config.js:5]: This should be 'node'
See: https://jestjs.io/docs/configuration#testenvironment-string

File: packages/byods/package.json [hunk 2]
```diff
@@ -25,10 +25,12 @@
...
     "fix:lint": "eslint 'src/**/*.ts' --fix",
     "fix:prettier": "prettier \"src/**/*.ts\" --write",
-    "prebuild": "rimraf dist"
+    "prebuild": "rimraf dist",
+    "test": "yarn test:style && yarn test:unit",
+    "test:style": "eslint 'src/**/*.ts'",
+    "test:unit": "jest --config=jest.config.js --runInBand"
   },
   "devDependencies": {
-    "@babel/preset-typescript": "7.16.7",
```
Comment (bhabalan) [inline packages/byods/package.json:31]: Why was this removed?

**Review by bhabalan — CHANGES_REQUESTED** [review]
Small comments

File: packages/byods/package.json (hunk 2 above)
Comment (Kesari3008) [reply packages/byods/package.json:31]: When we use Babel, we need this package to handle typescript files but I am using ts-jest to handle the TypeScript transpilation so this package doesn't seem to be required

File: packages/byods/jest.config.js (hunk 1 above)
Comment (Kesari3008) [reply packages/byods/jest.config.js:5]: I did use 'node' but while running the test I was receiving below error:

Comment (bhabalan) [reply packages/byods/jest.config.js:5]: Please go ahead and remove the import. We don't need it.

**Review by bhabalan — APPROVED** [review]

File: packages/byods/package.json [hunk 3]
```diff
@@ -25,7 +25,10 @@
     "docs": "typedoc --emit none",
     "fix:lint": "eslint 'src/**/*.ts' --fix",
     "fix:prettier": "prettier \"src/**/*.ts\" --write",
-    "prebuild": "rimraf dist"
+    "prebuild": "rimraf dist",
+    "test": "yarn test:style && yarn test:unit",
+    "test:style": "eslint 'src/**/*.ts'",
+    "test:unit": "jest --config=jest.config.js --runInBand"
```
Comment (Shreyas281299) [inline packages/byods/package.json:31]: Can we use webex-legacy-tools to run the tests?

File: packages/byods/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+export default {
```
Comment (Shreyas281299) [inline packages/byods/jest.config.js:1]: There is a jest config present that we use for all the packages can we import that and then add our additional config on top of it? To use the newer tooling.
```
const config = require('@webex/jest-config-legacy');

module.exports = {...config, 
         ourConfig: 'ourValue' ;
};
```

File: packages/byods/jest.d.ts
Comment (Shreyas281299) [inline packages/byods/jest.d.ts:1]: Why is this file required? I thought d.ts files are created when we build a package.

File: packages/byods/jest.setup.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    toBeCalledOnceWith(received, ...expected) {
+      try {
+        expect(received).toBeCalledTimes(1);
+        expect(received).toBeCalledWith(...expected);
+      } catch (error) {
+        return {message: () => error, pass: false};
+      }
+  
+      return {pass: true};
+    },
```
Comment (Shreyas281299) [inline packages/byods/jest.setup.js:11]: What does this file do?

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]

File: packages/byods/jest.setup.js (hunk 5 above)
Comment (Kesari3008) [reply packages/byods/jest.setup.js:11]: This I added to support custom jest matcher

File: packages/byods/jest.d.ts
Comment (Kesari3008) [reply packages/byods/jest.d.ts:1]: This is required for getting custom matchers to work. This is not the same as other d.ts files.

File: packages/byods/jest.config.js (hunk 4 above)
Comment (Kesari3008) [reply packages/byods/jest.config.js]: Addressed

**Review by Shreyas281299 — APPROVED** [review]
